prov/rxm/src/rxm_ep.c

```diff
@@ -410,16 +410,14 @@ static struct fi_ops_ep rxm_ops_ep = {
 	.tx_size_left = fi_no_tx_size_left,
 };
 
-/* Caller must hold recv_queue->lock. The lock will be released if we find a
- * matching message from the unexpected msg queue. */
+/* Caller must hold recv_queue->lock */
```

Comment:
Much better.  It is always preferable for the function that acquires a lock to release it, or locking becomes a mess.